File: Sort/Gold/BOJ_1715.cpp
#include <bits/stdc++.h>

using namespace std;

int n;
priority_queue<int> pq;
int result;
int main()
{
	cin >> n;
	while (n--)
	{
		int x;
		cin >> x;
		pq.push(-x);
	}
	
	while (pq.size() != 1)
	{
		int one = -pq.top();
		pq.pop();
		int two = -pq.top();
		pq.pop();

		int sum = one + two;
		result += sum;
		pq.push(-sum);
	}
	cout << result;



	return 0;
}

File: Sort/Gold/BOJ_1715.py
import heapq

n = int(input())

heap = []
for i in range(n):
  data = int(input())
  heapq.heappush(heap,data)

result = 0
while len(heap) != 1:
  one = heapq.heappop(heap)
  two = heapq.heappop(heap)
  sum_value= one + two
  result+=sum_value
  heapq.heappush(heap,sum_value)
  


print(result)



# 계속 작은 번호끼리 더하는게 핵심